vault/external_tests/api/kv_helpers_test.go

```diff
@@ -0,0 +1,169 @@
+package api
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	logicalKv "github.com/hashicorp/vault-plugin-secrets-kv"
+	"github.com/hashicorp/vault/api"
+	vaulthttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func TestKVHelpers(t *testing.T) {
+	t.Parallel()
+
+	// initialize test cluster
+	coreConfig := &vault.CoreConfig{
+		LogicalBackends: map[string]logical.Factory{
+			"kv":    logicalKv.Factory,
+			"kv-v2": logicalKv.VersionedKVFactory,
+		},
+	}
+
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	cores := cluster.Cores
+	core := cores[0].Core
+	client := cluster.Cores[0].Client
+	vault.TestWaitActive(t, core)
+
+	// mount the KVv2 backend
+	// (the test cluster has already mounted the KVv1 backend at "secret")
+	err := client.Sys().MountWithContext(context.Background(), "secret-v2", &api.MountInput{
+		Type: "kv-v2",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	secretData := map[string]interface{}{
+		"foo": "bar",
+	}
+
+	//// v1 ////
+	t.Run("kv v1 helpers", func(t *testing.T) {
+		if err := client.KVv1("secret").Put(context.Background(), "value", secretData); err != nil {
```

Comment:
This is pretty silly, I realize, but having the key name be `value` made me double take 😆.